LoopKitUI/Views/Settings Editors/TherapySettingsView.swift

```diff
@@ -267,8 +267,9 @@ extension TherapySettingsView {
     }
 
     private var basalRatesSection: Card {
-        card(for: .basalRate) {
-            if let schedule = viewModel.therapySettings.basalRateSchedule,
+        @State var basalRateSchedule = viewModel.therapySettings.basalRateSchedule // work-around for issue 2267 where basalRateSchedule doesn't update in the card
```

Comment:
Declaring @State inside a computed var is a misuse of the @State wrapper. It won't provide the state persistence expected because computed properties don't maintain storage between calculations. This might work coincidentally due to the onAppear update but this needs to be updated.

Reply:
Yes - I was aware, that’s why I explicitly commented that it’s a workaround (a proper fix would not need such a comment). I couldn’t find a reason why the current implementation stopped working correctly. If you have a cleaner way to fix then that’s better. Otherwise I preferred a benign workaround to the current buggy state.